src/main/java/org/assertj/core/api/recursive/comparison/FieldLocation.java

```diff
@@ -64,7 +63,15 @@ public boolean matches(String concatenatedPath) {
     return Stream.of(fieldPaths).map(FieldLocation::new).collect(toList());
   }
 
+  /**
+   * @deprecated use {@link #fieldLocation} instead
+   * */
+  @Deprecated
   public static FieldLocation fielLocation(String fieldPath) {
```

Comment:
If this class and especially this method can be `package-private`, then the renaming could be directly applied and the deprecation is not needed. Let's wait for the feedback from @joel-costigliola.

Reply:
Since the class is exposed through `RecursiveComparisonConfiguration`, I'm going to assume it is used, we can make `matches` package private as it was never meant to be exposed to users but the constructor and factory method should still be public.